### PR TITLE
Kanvi/config support II

### PR DESCRIPTION
**Pull request by kanvi-nervana**

### CODE REVIEW

File: src/ngraph_backend_manager.h [hunk 1]
```diff
@@ -65,6 +65,10 @@ class BackendManager {
 
   static void ReleaseBackend(const string& backend_name);
 
+  static void SetConfig(
+      const string& backend_name,
+      std::unordered_map<std::string, std::string> additional_attributes_map);
```
Comment (shresthamalik) [inline src/ngraph_backend_manager.h]: map can be made const & too

File: test/graph_rewrites/config_for_grappler_test.cc [hunk 2]
```diff
@@ -116,6 +115,73 @@ TEST(GrapplerConfig, RConfig1) {
...
+    node->set_requested_device("CPU");
+  }
+
+  // Create GraphDef and Grappler
+  grappler::GrapplerItem item;
+  graph.ToGraphDef(&item.graph);
+  ConfigProto config_proto;
+  auto backend_name = AttrValue();
+  backend_name.set_s("INTERPRETER");
+  auto device_config = AttrValue();
```
Comment (shresthamalik) [inline test/graph_rewrites/config_for_grappler_test.cc]: The device_config is not used right?

File: test/graph_rewrites/config_for_grappler_test.cc [hunk 3]
```diff
@@ -116,6 +115,73 @@ TEST(GrapplerConfig, RConfig1) {
...
+  // There is only one node in the graph
+  // And it is an NGraphEncapsulateOp
+  ASSERT_EQ(output_graph.num_op_nodes(), 1);
+  Node* ng_encap = nullptr;
+
+  for (auto node : output_graph.op_nodes()) {
+    ng_encap = node;
+  }
+  ASSERT_NE(ng_encap, nullptr);
+  string ng_backend, ng_device_config, ng_test_echo;
```
Comment (shresthamalik) [inline test/graph_rewrites/config_for_grappler_test.cc]: same comment here. We can remove it.

File: test/graph_rewrites/config_for_grappler_test.cc (hunk 2 above)
Comment (kanvi-nervana) [reply test/graph_rewrites/config_for_grappler_test.cc]: Right, missed removing it.

**Pull request merged**